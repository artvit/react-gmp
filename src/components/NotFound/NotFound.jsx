import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '../../shared/Dialog/dialog-buttons';
import Logo from '../../shared/logo/Logo';
import LogoBlock from '../../shared/logo/LogoBlock';
import { ButtonBlock, NotFoundTextBox } from './NotFound.style';

const NotFound = ({ onGoToHomeClick }) => (
  <>
    <LogoBlock><Logo size="20px" /></LogoBlock>
    <NotFoundTextBox>404</NotFoundTextBox>
    <ButtonBlock>
      <Button onClick={onGoToHomeClick}>Go Back To Home</Button>
    </ButtonBlock>
  </>
);

NotFound.propTypes = {
  onGoToHomeClick: PropTypes.func.isRequired
};

export default NotFound;
