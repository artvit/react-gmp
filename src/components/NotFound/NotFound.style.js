import styled from 'styled-components';

export const NotFoundTextBox = styled.div`
  padding: 70px 0 30px;
  width: 100%;
  text-align: center;
  font-size: 200px;
`;

export const ButtonBlock = styled.div`
  width: 100%;
  text-align: center;
  padding-bottom: 100px;
`;
