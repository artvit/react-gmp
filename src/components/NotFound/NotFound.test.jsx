import { render } from '@testing-library/react';
import React from 'react';

import NotFound from './NotFound';

describe('NotFound', () => {
  it('render component', () => {
    const { getByText, asFragment } = render(<NotFound onGoToHomeClick={jest.fn()} />);
    const element = getByText(/404/);
    expect(element).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
