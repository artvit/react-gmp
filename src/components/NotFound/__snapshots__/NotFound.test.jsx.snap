// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound render component 1`] = `
<DocumentFragment>
  <div
    class="sc-jSFkmK gjVGIP"
  >
    <span
      class="sc-hKFyIo iirmuI"
    >
      <span
        class="sc-eCApGN vYYqg"
      >
        netflix
      </span>
      roulette
    </span>
  </div>
  <div
    class="sc-gKAblj kxAosu"
  >
    404
  </div>
  <div
    class="sc-iCoHVE fwKHFs"
  >
    <button
      class="sc-gtssRu Fbbfx"
    >
      Go Back To Home
    </button>
  </div>
</DocumentFragment>
`;
